src/router.rs

```diff
@@ -27,14 +32,126 @@ impl ComponentRequest {
             method,
         }
     }
+}
+
+#[derive(Debug)]
+pub struct WorkerNode {
+    url: String,
+    components: Vec<String>,
+    is_active: bool,
+}
+
+impl WorkerNode {
+    pub fn new(url: String) -> Self {
+        Self {
+            url,
+            components: Vec::new(),
+            is_active: false,
+        }
+    }
+
+    pub fn get_component_list(&self) -> Result<Vec<String>, RouterError> {
+        let url = format!("{}/meta/status", self.url);
+        let body = reqwest::get(&url)?.text()?;
+
+        let response: StatusResponse = serde_json::from_str(&body)?;
+        debug!("{:?}", response.active_components);
+        let mut component_list = Vec::new();
+
+        for component in &response.active_components {
+            let component_name = format!("{}/{}", component.id.path.user, component.id.path.repo);
+            debug!("{}", component_name);
+            component_list.push(component_name);
+        }
+
+        debug!("{:?}", component_list);
+
+        Ok(component_list)
+    }
+}
+
+#[derive(Debug)]
+pub struct RequestForwarder {
+    workers: Vec<Arc<WorkerNode>>,
+    components_map: RwLock<HashMap<String, Arc<WorkerNode>>>,
+}
+
+impl RequestForwarder {
+    pub fn new() -> Result<Self, RouterError> {
+        let mut workers = Vec::new();
+        //Why does this not need to be mutable?
+        let components_map = RwLock::new(HashMap::new());
+
+        // TODO: load worker nodes from file into vector
+
+        //Worker One
+        workers.push(Arc::new(WorkerNode::new(String::from(
+            "http://ec2-34-228-212-219.compute-1.amazonaws.com",
+        ))));
+
+        //Worker Two
+        workers.push(Arc::new(WorkerNode::new(String::from(
+            "http://ec2-54-211-200-158.compute-1.amazonaws.com",
+        ))));
+
+        let res = Self {
+            workers,
+            components_map,
+        };
+
+        // - scan through server list to get active components
+        res.update_workers()?;
```

Comment:
If we just log on errors instead of returning the error, we can make this infallible. It's not that bad if the initial update fails anyway..